common/src/KokkosFFT_padding.hpp

```diff
@@ -107,240 +105,128 @@ void crop_or_pad_impl(const ExecutionSpace& exec_space, const InViewType& in,
-
-template <typename ExecutionSpace, typename InViewType, typename OutViewType>
-void crop_or_pad_impl(const ExecutionSpace& exec_space, const InViewType& in,
-                      OutViewType& out, shape_type<7> s) {
-  constexpr std::size_t DIM = 6;
-
-  auto [s0, s1, s2, s3, s4, s5, s6] = s;
-  out = OutViewType("out", s0, s1, s2, s3, s4, s5, s6);
-
-  int n0 = std::min(s0, in.extent(0));
-  int n1 = std::min(s1, in.extent(1));
-  int n2 = std::min(s2, in.extent(2));
-  int n3 = std::min(s3, in.extent(3));
-  int n4 = std::min(s4, in.extent(4));
-  int n5 = std::min(s5, in.extent(5));
-  int n6 = std::min(s6, in.extent(6));
-
-  using range_type = Kokkos::MDRangePolicy<
-      ExecutionSpace,
-      Kokkos::Rank<DIM, Kokkos::Iterate::Default, Kokkos::Iterate::Default>>;
-  using tile_type  = typename range_type::tile_type;
-  using point_type = typename range_type::point_type;
-
-  range_type range(exec_space, point_type{{0, 0, 0, 0, 0, 0}},
-                   point_type{{n0, n1, n2, n3, n4, n5}},
-                   tile_type{{4, 4, 4, 4, 1, 1}}
-                   // [TO DO] Choose optimal tile sizes for each device
-  );
-
-  Kokkos::parallel_for(
-      "KokkosFFT::crop_or_pad", range,
-      KOKKOS_LAMBDA(int i0, int i1, int i2, int i3, int i4, int i5) {
-        for (int i6 = 0; i6 < n6; i6++) {
-          out(i0, i1, i2, i3, i4, i5, i6) = in(i0, i1, i2, i3, i4, i5, i6);
-        }
-      });
-}
-
-template <typename ExecutionSpace, typename InViewType, typename OutViewType>
-void crop_or_pad_impl(const ExecutionSpace& exec_space, const InViewType& in,
-                      OutViewType& out, shape_type<8> s) {
-  constexpr std::size_t DIM = 6;
-
-  auto [s0, s1, s2, s3, s4, s5, s6, s7] = s;
-  out = OutViewType("out", s0, s1, s2, s3, s4, s5, s6, s7);
-
-  int n0 = std::min(s0, in.extent(0));
-  int n1 = std::min(s1, in.extent(1));
-  int n2 = std::min(s2, in.extent(2));
-  int n3 = std::min(s3, in.extent(3));
-  int n4 = std::min(s4, in.extent(4));
-  int n5 = std::min(s5, in.extent(5));
-  int n6 = std::min(s6, in.extent(6));
-  int n7 = std::min(s7, in.extent(7));
-
-  using range_type = Kokkos::MDRangePolicy<
-      ExecutionSpace,
-      Kokkos::Rank<DIM, Kokkos::Iterate::Default, Kokkos::Iterate::Default>>;
-  using tile_type  = typename range_type::tile_type;
-  using point_type = typename range_type::point_type;
-
-  range_type range(exec_space, point_type{{0, 0, 0, 0, 0, 0}},
-                   point_type{{n0, n1, n2, n3, n4, n5}},
-                   tile_type{{4, 4, 4, 4, 1, 1}}
-                   // [TO DO] Choose optimal tile sizes for each device
-  );
-
-  Kokkos::parallel_for(
-      "KokkosFFT::crop_or_pad", range,
-      KOKKOS_LAMBDA(int i0, int i1, int i2, int i3, int i4, int i5) {
-        for (int i6 = 0; i6 < n6; i6++) {
-          for (int i7 = 0; i7 < n7; i7++) {
-            out(i0, i1, i2, i3, i4, i5, i6, i7) =
-                in(i0, i1, i2, i3, i4, i5, i6, i7);
+  if constexpr (InViewType::rank() == 2) {
```

Comment:
Two related questions about this `crop_or_pad` function but not directly linked to this PR:

- Could we avoid manual unpacking of integers by using the conversion functions `Kokkos::Array` <-> `std::array` ?
- Why don't we use `Kokkos::deep_copy` with `Kokkos::subview` ?

Reply:
That is a good point. I will give it a try